### DISCUSSION TITLE
Question: Filter in get_event_list()

### DISCUSSION THREAD

**Discussion by faanskit** [SecKatie/wyzeapy#8] [Q&A]
Question:
When the Wyze app query for events from a camera with ./get_event_list, it asks for "all" events:
`    "event_value_list": [],
`

Wyzeapy sends a filter, that likely makes the app omit sound detections. Is this on purpose?
```
            "event_value_list": [
                "1",
                "13",
                "10",
                "12"
            ],
```

By adding "2" to this list, also sound events gets published to Home Assistant:
```
            "event_value_list": [
                "1",
                "2",
                "13",
                "10",
                "12"
            ],
```

Do you have documentation what 1, 13, 10 and 12 means?
1 = Camera Motion
2 = Sound Detection
10 - ?
12 - ?
13 - ?

**Comment by SecKatie** [comment]
Event value 13 is a Cam Plus event from my testing.

I don't know what 10 or 12 are.

**Reply by faanskit** [reply]
Ok, suggest to add also "2" for sound events from CAM and PAN.

**Comment by faanskit** [comment]
Following the discussion in this [comment](https://github.com/JoshuaMulliken/wyzeapy/issues/4#issuecomment-836764633), the thread continues here.

In order for this to be more flexible, I suggest the following approach:

1.
Add argument to get_events(),  get_latest_event() and get_event_list() of what types of events to be pulled.

Eg.
```
class EventTypes(Enum):
    SOUND = "Sound"
    MOTION = "Motion"
    ALL = "All"
```
2. Use input to device which events to pull

3. If no argument, use current implementation (MOTION)



The in ha-wyzeapi its yet another discussion. Should there be double sensors (motion and sound)?

**Reply by faanskit** [reply]
> In order for this to be more flexible, I suggest the following approach:

Yup, updated PR #5 accordingly. Fully backward compatible.
